### PR TITLE
Implemented savestates and quick resume

### PR DESCRIPTION
**Pull request by Jackamilio**
There's a frameskip bug that seems to occur though (sometimes just bringing up the menu without even interacting with savestates seems to trigger it)

### CODE REVIEW

File: src/pc/savestate.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#define SAVESTATE_EXCLUDE __attribute__((section("dontsave")))
+
+void savestate_check();
+
+void savestate_request_save(int slot);
+void savestate_request_load(int slot);
+
+void savestate_get_name(int slot, char* namebuffer);
+
+#endif
```
Comment (DrUm78) [inline src/pc/savestate.h]: Should have an empty line here.

File: src/pc/gfx/gfx_sdl_menu.h [hunk 2]
```diff
@@ -15,10 +15,10 @@ typedef enum{
...
 
 ///------ Definition of the different aspect ratios 
-// #define ASPECT_RATIOS \
-//     X(ASPECT_RATIOS_TYPE_STRETCHED, "STRETCHED") \
-//     X(ASPECT_RATIOS_TYPE_SCALED, "SCALED") \
-//     X(ASPECT_RATIOS_TYPE_CROPPED, "CROPPED") \
+// #define ASPECT_RATIOS 
+//     X(ASPECT_RATIOS_TYPE_STRETCHED, "STRETCHED") 
+//     X(ASPECT_RATIOS_TYPE_SCALED, "SCALED") 
+//     X(ASPECT_RATIOS_TYPE_CROPPED, "CROPPED")
```
Comment (DrUm78) [inline src/pc/gfx/gfx_sdl_menu.h]: Trailing spaces should be removed.

**Comment by DrUm78** [comment]
Seems good to me, I added  two comments just for nitpicking. :)
Also, maybe would be good to rename `src/pc/gfx/gfx_sdl2.c` to `src/pc/gfx/gfx_sdl.c` for more clarity as it was ported to SDL 1.2 for the FunKey S.

**Comment by DrUm78** [comment]
Also, as I said on Discord, the frameskip effect happened with the old code too, so not a regression here. Anwyay, it appears only once after closing the menu so not a big deal I guess.

**Comment by Jackamilio** [comment]
It should be ok now

**Comment by DrUm78** [comment]
Great work man, let's merge. 👍

**Pull request merged**